### PR TITLE
Anarchy runner pod management

### PR DESCRIPTION
**Pull request by jkupferer**
- Auto-scaling of AnarchyRunner pods
- Auto-restart of AnarchyRunner pods with repeated failures

### CODE REVIEW

**Review by rhjcd — APPROVED** [review]

File: api/anarchyrunner.py [hunk 1]
```diff
@@ -34,3 +37,24 @@ async def on_startup_running_all_in_one(cls):
         })
         await cls.cache_put(default)
         logging.info(f"Cache preloaded {default}")
+
+    async def update_status(self):
+        if Anarchy.running_all_in_one:
+            return
+        pods_status = []
+        for pod in anarchyrunnerpod.AnarchyRunnerPod.cache.values():
```
Comment (fridim) [inline api/anarchyrunner.py:45]: shouldn't this be `async for` ?

File: operator/anarchyrunner.py [hunk 2]
```diff
@@ -103,64 +134,105 @@ def make_pod_template(self, runner_token=None):
...
+        idle_runner_count = 0
+        for pod in self.status.get('pods', []):
+            if 'run' not in pod:
+                idle_runner_count += 1
+
+        if not have_pending_pod \
+        and self.scale_up_threshold \
+        and self.scale_up_threshold < len(self.status.get('pendingRuns', [])) - idle_runner_count \
+        and (not self.max_replicas or len(self.pods) < self.max_replicas) \
+        and (datetime.now(timezone.utc) - self.last_scale_up_datetime).total_seconds() > self.scale_up_delay:
```
Comment (fridim) [inline operator/anarchyrunner.py]: Should we simplify this condition by having plumbing methods?

```python
     if not have_pending_pod \
        and self.reached_threshold() \
        and not self.reached_max_replicas() \
        and self.passed_scale_up_delay():
```

And probably add unit tests for those methods to validate logic and edge cases.

**Review by fridim — APPROVED** [review]
